preview-env/create/action.yml

```diff
@@ -80,7 +80,14 @@ runs:
     shell: bash
     run: |-
       echo "::group::wait for sync"
-      argocd app wait ${{ inputs.app_name }} --timeout 1200 --health
+      i=1
+      until argocd app wait ${{ inputs.app_name }} --timeout 1200 --health \
```

Comment:
The `until` will retry the `argocd app wait ${{ inputs.app_name }} --timeout 1200 --health` whenever that exits with non-zero error code, if I understand correctly.

This can happen for multiple reasons (IIRC, we should double check):

- authentication problem (intended, this is what the PR wants to address)
- app doesn't become healthy within the timeout e.g. due to a persistent developer coding error (unintended, could lead also to cost increase since GHA workflow will run 5-times longer potentially)

Maybe there is a way to distinguish the two?

Reply:
Good point 👍 
ArgoCD seems to exit with a generic code (`20`), so `stderr` needs to be parsed for `code = PermissionDenied`.